do_ci.sh test_gcc CI target fails

*Title*: *do_ci.sh test_gcc CI target fails*

*Description*:
>When updating to [the latest envoy](https://github.com/envoyproxy/envoy/commit/0f0f675c22de944889298e0787e107265367b367), the `test_gcc` CI target fails with the following error.

```
E         [02:08:19.944607][11620][I] Nighthawk grpc service exits
E         pure virtual method called
E         terminate called without an active exception
```

https://dev.azure.com/cncf/envoy/_build/results?buildId=112587&view=logs&j=6d188cbb-0fce-5ab7-d058-3dafa1306ac4&t=3254ec1c-a4e0-51e2-2720-602bcbfef0a6

To not block the import, we have disabled these tests in https://github.com/envoyproxy/nighthawk/pull/869. We should reenable these tests ASAP.

*Reproduction steps*:
> Run do_ci.sh test_gcc in CI. Local test passes.

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.

*Logs*:
>Include the Nighthawk logs.

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.

*Call Stack*:
> If the Envoy binary is crashing, a call stack is **required**.
Please refer to the [Bazel Stack trace documentation](https://github.com/envoyproxy/envoy/tree/master/bazel#stack-trace-symbol-resolution).

--- a/.azure-pipelines/pipelines.yml
+++ b/.azure-pipelines/pipelines.yml
@@ -61,6 +61,23 @@ stages:
       parameters:
         ciTarget: $(CI_TARGET)
 
+- stage: test_gcc
+  dependsOn: ["check"]
+  pool: "x64-large"
+  jobs:
+  - job: test_gcc
+    displayName: "do_ci.sh"
+    strategy:
+      maxParallel: 1
+      matrix:
+        test_gcc:
+          CI_TARGET: "test_gcc"
+    timeoutInMinutes: 120
+    steps:
+    - template: bazel.yml
+      parameters:
+        ciTarget: $(CI_TARGET)
+
 - stage: sanitizers
   dependsOn: ["test"]
   pool: "x64-large"
@@ -102,6 +119,7 @@ stages:
 - stage: release
   dependsOn:
   - "clang_tidy"
+  - "test_gcc"
   - "sanitizers"
   - "coverage"
   condition: eq(variables['PostSubmit'], true)

--- a/source/client/service_impl.h
+++ b/source/client/service_impl.h
@@ -55,10 +55,10 @@ private:
   void writeResponse(const nighthawk::client::ExecutionResponse& response);
   grpc::Status finishGrpcStream(const bool success, absl::string_view description = "");
 
+  Envoy::Thread::MutexBasicLockable log_lock_;
   std::unique_ptr<Envoy::Logger::Context> logging_context_;
   std::shared_ptr<Envoy::ProcessWide> process_wide_;
   Envoy::Event::RealTimeSystem time_system_; // NO_CHECK_FORMAT(real_time)
-  Envoy::Thread::MutexBasicLockable log_lock_;
   grpc::ServerReaderWriter<nighthawk::client::ExecutionResponse,
                            nighthawk::client::ExecutionRequest>* stream_;
   std::future<void> future_;

--- a/source/client/service_impl.h
+++ b/source/client/service_impl.h
@@ -95,8 +95,8 @@ public:
                                nighthawk::request_source::RequestStreamRequest>* stream) override;
 
 private:
-  std::unique_ptr<Envoy::Logger::Context> logging_context_;
   Envoy::Thread::MutexBasicLockable log_lock_;
+  std::unique_ptr<Envoy::Logger::Context> logging_context_;
   RequestSourcePtr createStaticEmptyRequestSource(const uint32_t amount);
 };
 

